### DISCUSSION TITLE
LetsEncrypt certs not renewing

### DISCUSSION THREAD

**Discussion by ar0311** [ravendb/ravendb#14739] [General]
Hi,

I've just noticed my cluster cert's aren't automatically renewing.

I haven't seen any errors per se, just the studio notification that they are expiring soon (6 days)

The admin logs filtered for "certificate" when manually renewing within studio shows the following:

2022-08-12T02:23:10.7333613Z, 51, Information, Server, Raven.Server.RavenServerStartup, POST /admin/certificates/letsencrypt/force-renew - 307 - 0 ms
2022-08-12T02:23:11.1833559Z, 41, Operations, B, Raven.Server.Rachis.RachisConsensus, Received RecheckStatusOfServerCertificateReplacementCommand, index = 9688.
2022-08-12T02:23:11.1870812Z, 51, Operations, ServerStore, Raven.Server.ServerWide.ServerStore, The server certificate was successfully replaced in 1 nodes out of 3.

However, looking in /etc/ravendb/security shows the timestamps are not being updated on the certificate files and the certificate remains the same in studio.

Any idea where to start looking?

**Comment by gregolsky** [comment]
Hello,

The following endpoint returns a document containing information on the certificate replacement status - the state and confirmation count.

`GET /admin/certificates/replacement/status`

To reset the state and start fresh you need to make a *POST* request (supplying a client certificate) **on every node** to this endpoint:

`POST /admin/certificates/replacement/reset`

**Example using curl:**

For curl you'd first need to have your client certificate in the form of crt and key file

`curl -X POST --cert cert.crt --key cert.key https://{RAVEN_NODE_URL}/admin/certificates/replacement/reset`

Now the replacement status endpoint should return 404:

`GET /admin/certificates/replacement/status`

If you have done that **on all nodes**, you may start the renew process again.

1. Open the *Admin Logs* on **all nodes** to check for errors.
2. Go to the *Certificates* view in the Studio on one of the nodes
3. Click the *Renew* button.

If some nodes succeeded in replacing the file, but some didn't and we have a mixed state, it can be easier to just take the new certificate from the node that succeeded and copy it manually to the node which failed. Just replace the PFX file where configuration (Security.Certificate.Path) points to. The certificate file change should be picked up by the running server within one hour. You can also restart the service to make it use the new certificate immediately.

Please let me know if that helped.

**Reply by ar0311** [reply]
Thank you, following those steps allowed the renewal to complete successfully.
